astropy/coordinates/baseframe.py

```diff
@@ -273,9 +273,9 @@ def __init_subclass__(cls, **kwargs):
             if issubclass(basecls, BaseCoordinateFrame):
                 frame_attrs.update(basecls.frame_attributes)
 
-        for k, v in cls.__dict__.items():
-            if isinstance(v, Attribute):
-                frame_attrs[k] = v
+        frame_attrs |= {
+            k: v for k, v in cls.__dict__.items() if isinstance(v, Attribute)
```

Comment:
```suggestion
            k: v for k, v in vars(cls).items() if isinstance(v, Attribute)
```

Reply:
Using `vars` here would allow this to fit on one line.

Reply:
Quoting @mhvk from https://github.com/astropy/astropy/pull/16652

> Overall, my conclusion is to let this be -- quite a bit of churn for no tangible benefit.

I'd like to avoid going through another round of discussion if you don't mind too much, this *stylistic* PR has been stalling for a while already.

Reply:
We are editing this anyways, and editing it differently does not cause additional code churn.

Reply:
This code could be made much more compact by using `vars()`.

Reply:
looks like you *do* mind... I'll push this now if that's what it takes.